crates/codegen/parser_templates/src/shared/language.rs

```diff
@@ -0,0 +1,157 @@
+pub use std::ops::Range;
+pub use std::rc::Rc;
+
+#[allow(deprecated, unused_imports)]
+use semver::Version;
+
+pub use super::cst;
+pub use super::kinds::*;
+
+#[derive(PartialEq)]
+pub struct ParseError {
+    pub position: usize,
+    pub expected: String,
+}
+
+impl ParseError {
+    pub fn new<T: Into<String>>(position: usize, expected: T) -> Self {
+        Self {
+            position,
+            expected: expected.into(),
+        }
+    }
+
+    pub fn merge_with(&mut self, other: Self) {
+        // LEAVE THIS: it is super useful for debugging
```

Comment:
Is it possible to move the commented code to an uncommented function (and disable unused warning)?
This will make sure its contents survive future refactoring/moves, and stays up to date.

Reply:
Another pattern typically used in JS, is to add a file-level flag, so that you can just switch it on/off during debugging.

```rust
const IS_DEBUGGING: bool = false;

fn foo() {
    if IS_DEBUGGING {
        do_something();
    }

    do_another_thing();
}
```

Reply:
good idea.